### ISSUE TITLE
False negative when value is flush to colon in length-zero-no-unit

### ISSUE THREAD

**Issue by to2false** [stylelint/stylelint#2475]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
 
I remove space between attribute and value with no error under rule  `length-zero-no-unit: true`

> Which rule, if any, is this issue related to?

`length-zero-no-unit`

> What CSS is needed to reproduce this issue?

Get error
```css
.foo {
  margin: 0px;
}
```
Get no error
```css
.foo {
  margin:0px;
}
```
> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "length-zero-no-unit": true  
  }
}
```

> Which version of stylelint are you using?

`7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `npm run stylelint`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to SCSS maps."

> What did you expect to happen?

"Both cases get error."

**Comment by hudochenkov** [comment]
Thank you for the report! I can reproduce this problem in demo.

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2477
